tests/attention/test_attention_ts_context.py

```
@@ -1470,6 +1509,39 @@ def resolve(case: _PagedContextCase):
     )
 
 
+def test_attention_ts_context_paged_rejects_variable_window_before_compile(
+    monkeypatch,
+):
+    """Paged context must not silently dispatch variable windows as dense."""
+
+    with pytest.raises(
+        NotImplementedError,
+        match="variable-window masking.*not supported for paged context",
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)